java/client/src/main/java/glide/api/commands/StreamBaseCommands.java

```diff
@@ -105,4 +106,60 @@ public interface StreamBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> xdel(String key, String[] ids);
+
+    /**
+     * Reads entries from the given streams.
+     *
+     * @apiNote When in cluster mode, all keys in <code>keysAndIds</code> must map to the same hash
+     *     slot.
+     * @see <a href="https://redis.io/commands/xread/">redis.io</a> for details.
+     * @param keysAndIds A <code>Map</code> of keys and entry ids to read from. The <code>
+     *     Map</code> is composed of a stream's key and the id of the entry after which the stream
+     *     will be read.
+     * @return A <code>{@literal Map<String, Map<String, Map<String, String>>>}</code> with stream
```

Comment:
```suggestion
     * @return A <code>{@literal Map<String, Map<String, String[]>>}</code> with stream
```

Reply:
or
```suggestion
     * @return A <code>{@literal Map<String, Map<String, String[][]>>}</code> with stream
```
as per https://github.com/aws/glide-for-redis/pull/1494#issuecomment-2138450640
Please also describe that most inner String [2D] array is a map-like structure, which dislike of a `Map` can store duplicates.